packages/wonder-blocks-dropdown/components/multi-select.js

```diff
@@ -23,6 +23,17 @@ import {
 import typeof OptionItem from "./option-item.js";
 import type {DropdownItem} from "../util/types.js";
 
+type TranslatedLabels = {|
```

Comment:
I'd like to propose an alternative to what you've done here. What about if we just call this "labels" and make it so that people can use it for translations or just changing the text to something else.

Then it'd make the code a lot cleaner if we put all the strings in one location. For example in the constructor you can do something like: `Object.assign(labels, {clearSearch: "Clear search", ...}, props.labels)` - then later on you can just use the label directly.

Finally, if you do the above, you can make the type for the labels not only strict but also have all the labels be required: `type Labels = {| clearSearch: string, ... |};` In order to provide a property for the user to specify labels you can do: `$Shape<Labels>`, which means that they don't have to provide them all, just some of them.

Reply:
That makes a lot of sense to me. It feels more generic. Thanks for the suggestion!

Reply:
@jeresig As I'm making these changes, there's only one thing to have noticed. In the following scenario, we would need to keep in sync the default values:

### No custom `labels` provided

There will be cases where folks won't pass any custom labels to the `MultiSelect` component. When this happens, and due that some keys use dynamic values (number of total items, number of selected items, the selected item type has changed), we will need to keep them in sync.

These are the cases:

```
selectAllLabel: `Select all (${numOptions})`,
noneSelected: `0 ${props.selectItemType}`,
someSelected: `${props.selectedValues.length} ${props.selectItemType}`,
allSelected: `All ${props.selectItemType}`,
```

Maybe `selectItemType` won't change, but it can be a possibility. For the other cases

### Possible solution
We can achieve it by getting the computed values in `render`, `getDerivedStateFromProps` or `componentDidUpdate` (by checking against the prev and current values of each key):

e.g.
```
// if the user didn't provide any custom labels, then we need to compute the
current value internally, causing some extra re-renders
if (
    numCurrentOptions !== numPrevOptions &&
    !this.props.labels.selectAllLabel
) {
    this.setState({
        labels: {
            ...this.state.labels,
            selectAllLabel: `Select all (${numOptions})`,
        },
    });
}
```

What do you think about this case? any suggestions? Maybe we could still set the static labels in the constructor?

Reply:
@jandrade So this is highlighting a problem with the implementation of this, as it stands - if someone wants to implement a replacement for the selectAll label - how would that work? They would need access to the `numOptions` as well. I think the label definitions would have to look like this:

```
selectAll: ({numOptions: number, itemType: string}) => string,
allSelected: ({numOptions: number, itemType: string}) => string,
...
```

We could pass in any other properties that we'd expect people to use, as well. But this way they will have full control over the strings and how they are rendered and what is included. It also means that we can safely move the default values outside of the inline place where they're located.

Reply:
Got it. I was checking the translations and we are using a similar shape `(numOptions: number) => string`. I think that we could even get rid of the `itemType` prop.